lib/experimental/html/class-wp-html-tag-processor.php

```diff
@@ -1116,24 +1116,31 @@ private function after_tag() {
 	 * Converts class name updates into tag attributes updates
 	 * (they are accumulated in different data formats for performance).
 	 *
-	 * This method is only meant to run right before the attribute updates are applied.
-	 * The behavior in all other cases is undefined.
-	 *
 	 * @return void
 	 * @since 6.2.0
 	 *
 	 * @see $classname_updates
 	 * @see $lexical_updates
 	 */
 	private function class_name_updates_to_attributes_updates() {
-		if ( count( $this->classname_updates ) === 0 || isset( $this->lexical_updates['class'] ) ) {
+		if ( count( $this->classname_updates ) === 0 ) {
 			$this->classname_updates = array();
 			return;
 		}
 
-		$existing_class = isset( $this->attributes['class'] )
-			? substr( $this->html, $this->attributes['class']->value_starts_at, $this->attributes['class']->value_length )
-			: '';
+		if ( isset( $this->lexical_updates['class'] ) ) {
+			$existing_class_attr  = trim( $this->lexical_updates['class']->text );
+			$existing_class_value = substr( $existing_class_attr, strlen( 'class' ) + 2, -1 );
```

Comment:
There's also a (silly) case of `$p->set_attribute('class', true);` where `$existing_class_attr` will only contain `class`.

Reply:
Test coverage in https://github.com/WordPress/gutenberg/pull/46680/commits/4c225277de5bdaddf5d95a3164fbbe523ec2b070, fix in https://github.com/WordPress/gutenberg/pull/46680/commits/c48ba893e92a44b3a43691c7905fdf4422a0ab74.